libcxx/test/std/numerics/c.math/hermite.pass.cpp

```diff
@@ -26,7 +26,13 @@
 
 #include "type_algorithms.h"
 
-inline constexpr unsigned g_max_n = 128;
+template <class Real>
+constexpr unsigned get_maximal_order() {
+  if constexpr (std::numeric_limits<Real>::max_exponent10 < std::numeric_limits<Real>::max_exponent)
```

Comment:
This might not test what we actually need. 
My idea was rather along the lines of max_exponent10< 60 then return 39 else if max_exponent10 <... return 128. Hope this helps otherwise let me know and I can reformulate this. 
We should check which exponents we actually need here (e.g. the 60 and ... above).

(Sorry for the formatting. I am on my phone.)

Reply:
I had something along the lines of "Compute maximal order s.t. polynomials do not exceed maximal representable number (i.e. 10^(max_exponent10) )" in mind. But it gets quite messy as we would need to distinguish between floats that have NaNs/Infs and not.

Alternative that should suffice here: Use something similar to your original implementation without the preprocessor checks.

```cpp
template <class Real>
constexpr unsigned get_maximal_order() {
  if constexpr (std::numeric_limits<Real>::is_iec559)
    return 128;
  else { // Workaround for IBM hexadecimal floating-points.
    // Note $|H_n(x)| < 10^75$ for $n < 39$ and x in sample_points().
    static_assert(std::numeric_limits<Real>::max_exponent10 == 75);
    return 39;
  }
}
```

Reply:
Agree, thank you for suggestions.